uktrade/data-hub-frontend
# Add code coverage to our functional test suite

## Description of change

The intent of this PR is to capture code coverage for our functional tests. The reports generated by this PR will then be used in another job to calculate total test coverage.

Bonus: code coverage also captured on our e2e tests

## Test instructions

All tests should pass as usual (CI) and nyc reports should be saved in artifacts for each functional test suite instance.

For local, run the following steps:

1:. Start sandbox
2:. Build the application running command: `yarn build`
3:. Start the application (ensure NODE_ENV is `development` or `test`) running command: `yarn start:coverage`
4:. Start cypress and execute any test you wish by running command: `yarn test:functional:watch`

At the end of the test, look at `projectroot/cypress-coverage/lcov-report/index.html` and look for the files under test in order to view coverage.

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"
